src/SharpGLTF.Cesium/BinaryTable.cs

```diff
@@ -0,0 +1,208 @@
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Y), 0, result, i * 8 + 4, 4);
+            }
+            return result;
+        }
+
+        private static byte[] Vector3ToBytes<T>(IReadOnlyList<T> values)
+        {
+            var result = new byte[values.Count * 12];
+            for (int i = 0; i < values.Count; i++)
+            {
+                var vec = (Vector3)(object)values[i];
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.X), 0, result, i * 12, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Y), 0, result, i * 12 + 4, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Z), 0, result, i * 12 + 8, 4);
+            }
+            return result;
+        }
+
+        private static byte[] Vector4ToBytes<T>(IReadOnlyList<T> values)
+        {
+            var result = new byte[values.Count * 16];
+            for (int i = 0; i < values.Count; i++)
+            {
+                var vec = (Vector4)(object)values[i];
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.X), 0, result, i * 16, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Y), 0, result, i * 16 + 4, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Z), 0, result, i * 16 + 8, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.W), 0, result, i * 16 + 12, 4);
+            }
+            return result;
+        }
+
+        public static List<int> GetStringOffsets(List<string> values)
+        {
+            var offsets = new List<int>() { 0 };
+            foreach (var value in values)
+            {
+                var length = Encoding.UTF8.GetBytes(value).Length;
+                offsets.Add(offsets.Last() + length);
+            }
+
+            return offsets;
+        }
+
+
+        public static List<int> GetStringOffsets(List<List<string>> values)
+        {
+            var offsets = new List<int>() {};
+            foreach (var arr in values)
+            {
+                var arrOffsets = GetStringOffsets(arr);
+                var last = offsets.LastOrDefault();
+                foreach (var offset in arrOffsets)
+                {
+                    if(!offsets.Contains(last + offset))
+                    {
+                        offsets.Add(last + offset);
+                    }
+                }
+            }
+
+            return offsets;
+        }
+
+
+        public static List<int> GetArrayOffsets<T>(List<List<T>> values)
+        {
+            var offsets = new List<int>() { 0 };
+            foreach (var value in values)
+            {
+                offsets.Add(offsets.Last() + value.Count);
+
+            }
+            return offsets;
+        }
+
+        public static int GetSize<T>()
+        {
+            var isValueType = typeof(T).IsValueType;
+            Guard.IsTrue(isValueType, nameof(T), "T must be a value type");
+
+            var type = typeof(T);
+            int size = Marshal.SizeOf(Activator.CreateInstance(type));
```

Comment:
Activators are fishy because they don't go well with Trimmable/AOT.

use ```Marshal.SizeOf<T>();```

Reply:
Fixed in https://github.com/vpenades/SharpGLTF/pull/206/commits/187e405b3745ab3b32c2e3b280843ce6b2a2d5ed